src/test/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/CaseEventTaskHandlerTest.java

```diff
@@ -56,34 +61,58 @@ void init() {
+            .build();
+
+        CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+
+        when(coreCaseDataService.startUpdate(eq(CASE_ID), eq(NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE)))
+            .thenReturn(StartEventResponse.builder().caseDetails(caseDetails).build());
+
+        when(coreCaseDataService.submitUpdate(eq(CASE_ID), any(CaseDataContent.class))).thenReturn(caseData);
+
+        caseEventTaskHandler.execute(mockExternalTask, externalTaskService);
+
+        verify(coreCaseDataService).startUpdate(eq(CASE_ID), eq(NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE));
+        verify(coreCaseDataService).submitUpdate(eq(CASE_ID), any(CaseDataContent.class));
+        verify(externalTaskService).complete(mockExternalTask);
+    }
+
+    @Test
+    void shouldCallHandleFailureMethod_whenExceptionFromBusinessLogic() {
+        String errorMessage = "there was an error";
+
+        when(mockExternalTask.getRetries()).thenReturn(null);
+        when(coreCaseDataService.startUpdate(CASE_ID, NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE))
+            .thenAnswer(invocation -> {
+                throw new Exception(errorMessage);
+            });
 
-        @Test
-        void shouldTriggerCCDEvent_whenHandlerIsExecuted() {
-            CaseData caseData = new CaseDataBuilder().atStateClaimDraft()
-                .businessProcess(BusinessProcess.builder().status(BusinessProcessStatus.READY).build())
-                .build();
+        caseEventTaskHandler.execute(mockExternalTask, externalTaskService);
 
-            CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+        verify(externalTaskService, never()).complete(mockExternalTask);
+        verify(externalTaskService).handleFailure(mockExternalTask, "worker", errorMessage, 2, 500L);
+    }
 
-            when(coreCaseDataService.startUpdate(eq(CASE_ID), eq(NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE)))
-                .thenReturn(StartEventResponse.builder().caseDetails(caseDetails).build());
+    @Test
+    void shouldNotCallHandleFailureMethod_whenExceptionOnCompleteCall() {
+        String errorMessage = "there was an error";
 
-            when(coreCaseDataService.submitUpdate(eq(CASE_ID), any(CaseDataContent.class))).thenReturn(caseData);
+        doThrow(new NotFoundException(errorMessage)).when(externalTaskService).complete(mockExternalTask);
 
-            caseEventTaskHandler.execute(mockExternalTask, externalTaskService);
+        caseEventTaskHandler.execute(mockExternalTask, externalTaskService);
 
-            verify(coreCaseDataService).startUpdate(eq(CASE_ID), eq(NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE));
-            verify(coreCaseDataService).submitUpdate(eq(CASE_ID), any(CaseDataContent.class));
-            verify(externalTaskService).complete(mockExternalTask);
-        }
+        verify(externalTaskService, never()).handleFailure(mockExternalTask, "worker", errorMessage, 3, 500L);
```

Comment:
We should be as generic as possible when verifying no interactions, so `any()` should be used instead of specific arguments.

Reply:
I agree, I have done this elsewhere, must have missed this one 👍